app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/extensions/NetworkExtension.kt

```diff
@@ -0,0 +1,71 @@
+@file:Suppress("UnusedPrivateMember")
+
+package com.kks.nimblesurveyjetpackcompose.util.extensions
+
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
+import com.kks.nimblesurveyjetpackcompose.model.response.CustomErrorResponse
+import com.squareup.moshi.JsonDataException
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import kotlinx.coroutines.*
+import retrofit2.HttpException
+import retrofit2.Response
+import java.io.IOException
+import java.net.UnknownHostException
+
+private const val API_WAIT_TIME = 7000L
+const val UNKNOWN_ERROR = "Unknown error"
+const val SUCCESS_WITH_NULL_ERROR = "Success with null error"
+const val UNKNOWN_ERROR_MESSAGE = "Unknown error message"
+const val NETWORK_ERROR = "Network Error"
+
+const val EMAIL_EMPTY_MESSAGE = "Email cannot be empty"
+const val PASSWORD_EMPTY_MESSAGE = "Password cannot be empty"
+const val SUCCESS_MESSAGE = "success"
```

Comment:
Just a heads-up, the error messages should be handled from the Activity/Fragment and should be localized when displayed to the user.